Authenticator/Source/TableDiff.swift

```diff
@@ -31,92 +31,63 @@ protocol Identifiable {
 
 enum Change {
     case Insert(index: Int)
-    case Update(index: Int)
+    case Update(oldIndex: Int, newIndex: Int)
     case Delete(index: Int)
     // TODO: Consolidate matching Inserts and Deletes into Moves
     case Move(fromIndex: Int, toIndex: Int)
 }
 
-func changesFrom<T>(oldArray: [T], to newArray: [T], hasSameIdentity: (T, T) -> Bool) -> [Change] {
-    return changes(from: ArraySlice(oldArray), to: ArraySlice(newArray),
-        comparator: hasSameIdentity, equation: { (_, _) in false })
+func changesFrom<T: Identifiable>(oldItems: [T], to newItems: [T]) -> [Change] {
+    return changesFrom(oldItems, to: newItems, hasSameIdentity: { $0.hasSameIdentity($1) },
+        isEqual: { (_, _) in false })
 }
 
-func changesFrom<T: Identifiable where T: Equatable>(oldArray: [T], to newArray: [T]) -> [Change] {
-    return changes(from: ArraySlice(oldArray), to: ArraySlice(newArray),
-        comparator: { $0.hasSameIdentity($1) }, equation: ==)
+func changesFrom<T: Identifiable where T: Equatable>(oldItems: [T], to newItems: [T]) -> [Change] {
+    return changesFrom(oldItems, to: newItems, hasSameIdentity: { $0.hasSameIdentity($1) },
+        isEqual: ==)
 }
 
-private func changes<Row>(from oldRows: ArraySlice<Row>, to newRows: ArraySlice<Row>,
-    comparator rowsHaveSameIdentity: (Row, Row) -> Bool, equation rowsAreEqual: (Row, Row) -> Bool)
-    -> [Change]
+// Diff algorithm from the Eugene Myers' paper "An O(ND) Difference Algorithm and Its Variations"
+private func changesFrom<T>(oldItems: [T], to newItems: [T], hasSameIdentity: (T, T) -> Bool,
+    isEqual: (T, T) -> Bool) -> [Change]
```

Comment:
Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)